data/src/main/java/hera/store/unit/StorageAccessUnit.java

```diff
@@ -38,11 +46,32 @@ public void updateStore() {
 
 	public void add(M object) {
 		try {
-			dao.insert(object);
+			retryOnFail(() -> dao.insert(object));
 			data.add(object);
 		} catch(Exception e) {
 			LOG.error("Error while trying to add entity of type {}", entityName);
 			LOG.debug("Stacktrace:", e);
 		}
 	}
+
+	public void retryOnFail(Runnable runnable) throws Exception {
+		for(int i = 0; i < 3; i++) {
+			try {
+				runnable.run();
+				return;
+			} catch (Exception e) {
```

Comment:
might wanna check here that it only works for no connection exception since you dont wanna retry bad calls 3 times

Reply:
wont change for now since logging is sufficent